src/backend/common/models/alliance.py

```diff
@@ -11,7 +11,7 @@
 class _EventAllianceOptional(TypedDict, total=False):
     declines: List[TeamKey]
     name: str
-    backup: EventAllianceBackup
+    backup: Optional[EventAllianceBackup]
```

Comment:
cc @phil-lopreiato This ended up being the big change here - we store `backup` as `null` in the database if it doesn't exist, which seems right? I'm unsure about this one.

Reply:
We were omitting it from the structure before if there was no backup (hence the `total=False` part), although maybe the parser was explicitly setting `None`? I suppose as long as we have all the cases null checks, which the type checker should enforce, this is an okay change

Reply:
Oh wild. Okay I get this. Lemme take another pass and see if I can do this without making it optional.

Reply:
Yeah okay - I forget my context but I probably modeled it the same way the `event_team_status` https://github.com/the-blue-alliance/the-blue-alliance/blob/d9131246e3bddb96979b18610068b43e9e7d0aaa/src/backend/common/models/event_team_status.py#L65

Reply:
We could also store nulls - it would be fine (assuming the rest of the code passes the type check too)

Reply:
I'm good - we'll keep this as a non-Optional for now. I was trying to remember how I wound up where I did.

https://github.com/the-blue-alliance/the-blue-alliance/pull/3432/files#diff-b65753f8e7bb8f759942deb3731da998d5541339cd9d45675c07baac8cbe74f9R48